R/generateaWhereChart.R

```diff
@@ -79,514 +79,580 @@ generateaWhereChart <- function(data
+                     ,y = ymin
+                     ,colour = SD_label)
+                ,alpha = 0
+                ,linetype = "blank") 
+    
+    nRowsFill <- 2
+  }
+  
+  #include SD info for main variable
+  if (includeSTD == TRUE) {
+    chart <- 
+      chart + 
+      geom_ribbon(data = chart_data_long[[1]]
+                  ,aes(x = date
+                       ,ymin = ymin
+                       ,ymax = ymax
+                       ,fill = SD_label)
+                  ,alpha = 0.3 # adjust transparency of SD DEV shading
+                  ,linetype = "blank") 
+  }
+  
+  #add in line charts for other variables
+  #linetypes <- c("solid", "dotted", "dashed")
+  if (length(chart_data_long) > 1) {
+    for (x in 2:length(chart_data_long)) {
+      chart <-
+        chart +
+        geom_line(data = chart_data_long[[x]]
+                  ,aes(x = date
+                       ,y = measure/scalingFactor[[x]]
+                       ,colour = Variable)
+                  ,size = line_width
+                  #,linetype = linetypes[x] # change line type for sunsequent variables
+        ) +
+        scale_y_continuous(sec.axis = sec_axis(~.*scalingFactor[[x]]
+                                               ,name = ylabel[[x]])) 
     }
+  }
+  
+  #format figure
+  chart <- 
+    chart +
+    colorScaleToUse +
+    colorFillToUse +
+    # plot theme: https://ggplot2.tidyverse.org/reference/ggtheme.html
+    theme_bw() + 
+    
+    # format the text color, size, angle, and face for x- and y- axes. 
+    # x-axis labels 
+    theme(axis.text.x = element_text(color = "grey20", # font color 
+                                     size = size_font_axis_labels, # font size 
+                                     angle = 45,       # font angle 
+                                     hjust = 1,        # horizontal adjustment
+                                     face = "plain"),  # font type "plain", "bold" 
+          # y-axis labels 
+          axis.text.y = element_text(color = "grey20", 
+                                     size = size_font_axis_labels, 
+                                     face = "plain"),
+          # y-axis titles 
+          axis.title.y = element_text(color = "grey20", 
+                                      size = size_font_axis_titles, 
+                                      face = "bold"),
+          # turn off the x-axis title 
+          axis.title.x=element_blank()) + 
+    
+    # format the legend 
+    theme(#legend.position = "bottom" # place legend at bottom of chart
+      legend.position="right" # place legend at right side of chart
+      #,legend.box = "horizontal"
+      ,legend.box = "vertical"
+      #,legend.direction= "horizontal"
+      ,legend.direction= "vertical"
+      ,legend.title = element_blank() # no legend title
+      ,legend.justification = "center"
+      ,legend.text=element_text(size=size_font_legend_entries)) +
     
-    return(chart)
-}
+    # Reorder legend entries
```

Comment:
As well as the guide logic - this is an artifact from testing

Reply:
no, I think it should return the object so the user can do something with it if they desire

Reply:
Agreed - I was just referring to the legend guides parameters with this comment